plugins/bunch_o_snapshots/plugin.js

```diff
@@ -0,0 +1,416 @@
+          confirm: 1,
+          cancel: 0,
+          title: "All screenshots created!",
+          message: `Paths for screenshots:\n${allSaves
+            .map((saved) => `<br>- ${saved}`)
+            .join("")}`,
+          icon: "icon-saved",
+          width: electron.getCurrentWindow().getContentBounds().width / 2,
+        });
+      }
+
+      function takeScreenshot() {
+        if (currentSettings.devMode !== false) return;
+        let savePath;
+        options = {
+          width: 320,
+          height: 320,
+        };
+
+        let preview = main_preview;
+        let prevTarget = new THREE.Vector3();
+        preview.camera.getWorldDirection(prevTarget);
+        const oldData = {
+          isOrtho: preview.isOrtho,
+          axis: preview.camOrtho.axis,
+          position: preview.camera.position.clone(),
+          target: prevTarget,
+        };
+        const captureData = {
+          isOrtho: true,
+          axis: null,
+          position: new THREE.Vector3(-1, 1, -1),
+          target: new THREE.Vector3(0, 0, 0),
+        };
+        function rollbackPreview() {
+          preview.isOrtho = oldData.isOrtho;
+          preview.camOrtho.axis = oldData.axis;
+          preview.camera.position.copy(oldData.position);
+          preview.camera.lookAt(oldData.target);
+        }
+        function fixPreview() {
+          preview.isOrtho = captureData.isOrtho;
+          preview.camOrtho.axis = captureData.axis;
+          preview.camera.position.copy(captureData.position);
+          preview.camera.lookAt(captureData.target);
+          preview.resize();
+        }
+
+        let cb = null;
+        fixPreview();
+        Canvas.withoutGizmos(function () {
+          preview.render();
+
+          let frame = new CanvasFrame(preview.canvas);
+          frame.autoCrop();
+
+          if (options.width && options.height) {
+            let new_frame = new CanvasFrame(options.width, options.height);
+            let width = frame.width;
+            let height = frame.height;
+            if (width > options.width) {
+              height /= width / options.width;
+              width = options.width;
+            }
+            if (height > options.height) {
+              width /= height / options.height;
+              height = options.height;
+            }
+            new_frame.ctx.drawImage(
+              frame.canvas,
+              (options.width - width) / 2,
+              (options.height - height) / 2,
+              width,
+              height
+            );
+            frame = new_frame;
+          }
+
+          let dataUrl = frame.canvas.toDataURL();
+          let img = new Image();
+          img.src = dataUrl;
+          img.className = "allow_default_menu checkerboard";
+          const projectName = Project
+            ? Project.name.replace(/\.geo$/, "")
+            : "screenshot";
+          savePath = `${currentSettings.outputDirectory}\\${projectName}.png`;
```

Comment:
The backslash for file paths is only going to work on Windows. Please use the character that works for the respective system. You can use the NodeJS path module (`PathModule`) to make this easier and more reliable.